mlrun/feature_store/feature_set.py

```diff
@@ -384,15 +387,28 @@ def upsert_feature(name):
             aggregations = state.class_args.get("aggregates", [])
             aggregations.append(aggregation)
             state.class_args["aggregates"] = aggregations
+            if emit_policy:
+                state.class_args["emit_policy"] = emit_policy
         else:
-            state = graph.add_step(
-                name=state_name,
-                after=after or previous_step,
-                before=before,
-                class_name="storey.AggregateByKey",
-                aggregates=[aggregation],
-                table=".",
-            )
+            if emit_policy:
+                state = graph.add_step(
+                    name=state_name,
+                    after=after or previous_step,
+                    before=before,
+                    class_name="storey.AggregateByKey",
+                    aggregates=[aggregation],
+                    table=".",
+                    emit_policy=emit_policy,
+                )
+            else:
+                state = graph.add_step(
+                    name=state_name,
+                    after=after or previous_step,
+                    before=before,
+                    class_name="storey.AggregateByKey",
+                    aggregates=[aggregation],
+                    table=".",
+                )
```

Comment:
Too much code duplication, IIUC can be replaced to something like:
```suggestion
            class_args = {}
            if emit_policy:
                class_args["emit_policy"] = emit_policy
            state = graph.add_step(
                name=state_name,
                after=after or previous_step,
                before=before,
                class_name="storey.AggregateByKey",
                aggregates=[aggregation],
                table=".",
                **class_args,
            ) 
```